src/NHibernate/Linq/Visitors/ParameterTypeLocator.cs

```diff
@@ -110,6 +111,12 @@ public static class ParameterTypeLocator
 							out _,
 							out _))
 						{
+							if (visitor.SequenceSelectorExpressions.Contains(memberExpression) && type is IAssociationType associationType)
```

Comment:
Better to use `IsAssociationType` instead of casting for type check:
```C#
if (type.IsAssociationType && visitor.SequenceSelectorExpressions.Contains(memberExpression))
```

Reply:
Done.